### PR TITLE
Fix #3577: Add jest unit testing to the Divider component

### LINKED ISSUE #3577: Divider: Add jest unit testing to the Divider component
### Describe the feature you would like to see added

Add jest unit testing to the Divider component

### Is your feature request related to a problem?

_No response_

### Describe the solution you'd like

_No response_

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by habubey**
### Feature Requests
Add jest unit testing to the Divider component

### CODE REVIEW

File: components/lib/divider/Divider.spec.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import '@testing-library/jest-dom';
+import { render } from '@testing-library/react';
+import { Divider } from './Divider';
+
+describe('Divider', () => {
+    test('when tpe has not any property it returns with default class', () => {
```
Comment (melloware) [inline components/lib/divider/Divider.spec.js:6]: `when component has no properties it returns with default class`

File: components/lib/divider/Divider.spec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        // Arrange
+        const { container } = render(<Divider />);
+
+        // Act + Assert
+        expect(container.getElementsByClassName('p-divider p-component p-divider-horizontal').length).toBe(1);
```
Comment (melloware) [inline components/lib/divider/Divider.spec.js:11]: A better way to test for this is
```js
 expect(container).toHaveClass('p-divider p-component p-divider-horizontal');
```

File: components/lib/divider/Divider.spec.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    });
+    test('when layout and align as property it returns with class', () => {
+        // Arrange
+        const { container } = render(<Divider layout={'vertical'} align={'bottom'} />);
+
+        // Act + Assert
+        expect(container.getElementsByClassName('p-divider p-component p-divider-vertical p-divider-bottom').length).toBe(1);
+    });
+
+    test('when type has not any property it returns with default class', () => {
```
Comment (melloware) [inline components/lib/divider/Divider.spec.js:64]: `when type has no property it returns with the default class`

File: components/lib/divider/Divider.spec.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        // Arrange
+        const { container } = render(<Divider />);
+
+        // Act + Assert
+        expect(container.getElementsByClassName('p-divider p-component p-divider-solid').length).toBe(1);
+    });
+    test('when type as property it returns with class', () => {
```
Comment (melloware) [inline components/lib/divider/Divider.spec.js:71]: `when type is dashed it is styled as a dashed line`

File: components/lib/divider/Divider.spec.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        // Arrange
+        const { container } = render(<Divider type={'dashed'} />);
+
+        // Act + Assert
+        expect(container.getElementsByClassName('p-divider p-component p-divider-dashed').length).toBe(1);
+    });
+    test('when type as property it returns with class', () => {
```
Comment (melloware) [inline components/lib/divider/Divider.spec.js:78]: `when type is dotted is is styled as a dotted line`

File: components/lib/divider/Divider.spec.js (hunk 2 above)
Comment (melloware) [reply components/lib/divider/Divider.spec.js:11]: I would make this change throughout this whole test

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/divider/Divider.spec.js (hunk 2 above)
Comment (kalinkrustev) [reply components/lib/divider/Divider.spec.js]: Have you considered using https://jestjs.io/docs/snapshot-testing ?
I think it is more convenient, as the test code is separate from the expected output, so you do not need to maintain these classes in the test files. When something changes, it is usually more clear what the changes are when observing them in the snapshot.

Comment (bahadirsofuoglu) [reply components/lib/divider/Divider.spec.js]: Hi. I don't think it's a UI control. It tests the effect of certain logic on the UI and gets help from classes. For this reason, I could not associate it with snapshot tests. But if you'd like to send a sample, we'd be happy to review it.

Comment (kalinkrustev) [reply components/lib/divider/Divider.spec.js]: OK, I will send you an example what I mean. It is capturing the generated markup and making sure it does not change unexpectedly.

Comment (kalinkrustev) [reply components/lib/divider/Divider.spec.js]: Check here: #3579

**Pull request merged**